### PR TITLE
[PE-4640] Fix font-size bug in text utility

### PR DESCRIPTION
**Pull request by jllr**
https://ctl.atlassian.net/browse/PE-4640

Had issues with !important declaration in h2 and h3 tags in two components.
Modal and popover components classes: chi-popover__title, chi-modal__title and chi-modal__subtitle are wrapped under h2 and h3 tags.
In order to solve that, I had to set !important in these classes. Another solution would be to remove h2 and h3 tags and let them wrapped into divs.
@mattnickles @janibolkvadze thoughts?

### CODE REVIEW

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://0ada72e65169ea4ada1fad8858285e4379b96b48-chi-pr.enterprise.ctl.io" target="_blank"> https://0ada72e65169ea4ada1fad8858285e4379b96b48-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/857/ for more details.

File: src/chi/foundations/typography/typography.scss [hunk 1]
```diff
@@ -35,51 +35,51 @@ a {
 }
 
 h1,
-.chi-h1 {
-  font-size: 3rem;
-  font-weight: 700;
-  line-height: 4rem;
-  margin: 2rem 0;
+.-text--h1 {
```
Comment (mattnickles) [inline src/chi/foundations/typography/typography.scss:38]: @jllr @janibolkvadze  !important should only be defined on the text utility class, not the heading elements

Comment (mattnickles) [reply src/chi/foundations/typography/typography.scss:38]: Heading tag styles are default rules, text utility styles are customization rules. Would be clearer if we leave default heading styles in typography and move text utility styles to utilities/text.

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://6312fe1e385a2648638b54be6988c50fbc556da8-chi-pr.enterprise.ctl.io" target="_blank"> https://6312fe1e385a2648638b54be6988c50fbc556da8-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/861/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
The following instances are being terminated: https://ops.elasticbox.com/services/instances/i-jqvz3y, https://ops.elasticbox.com/services/instances/i-970wv1

**Pull request merged**